kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -180,13 +180,13 @@ maybeTermEquals notSimplifier childTransformers first second = do
         | Just boolAndData <- Builtin.Bool.matchUnifyBoolAnd second first =
             lift $ Builtin.Bool.unifyBoolAnd childTransformers second boolAndData
         | Just boolOrData <- Builtin.Bool.matchUnifyBoolOr first second =
-            lift $ Builtin.Bool.unifyBoolOr childTransformers second boolOrData
-        | Just boolOrData <- Builtin.Bool.matchUnifyBoolOr second first =
             lift $ Builtin.Bool.unifyBoolOr childTransformers first boolOrData
+        | Just boolOrData <- Builtin.Bool.matchUnifyBoolOr second first =
+            lift $ Builtin.Bool.unifyBoolOr childTransformers second boolOrData
```

Comment:
My comment here this morning was that we should modify `matchUnifyBoolOr` so that the required pattern (`first` or `second`) is included in `boolOrData`. That way, we don't have this coordination problem between `AndTerms` and `matchUnifyBoolOr`. But that is something @emarzion 's pull request.

Reply:
My pull request does handle this now (https://github.com/kframework/kore/blob/0503e7e35f64a80b3132e5641ea2dd4f024c57ff/kore/src/Kore/Step/Simplification/AndTerms.hs#L178) though `first, second` remain as arguments.  Should I place those in the records as well?

Reply:
@emarzion What do those arguments represent?

Reply:
Those are the two terms being unified.